src/otk_external_osbuild/command/gen_depsolve_dnf4.py

```diff
@@ -0,0 +1,101 @@
+import hashlib
+import json
+import os
+import subprocess
+import sys
+
+
+def transform(packages):
+    """Transform the output of `osbuild-depsolve-dnf4` to the output format
+    expected of this external."""
+
+    data = {"tree": {"const": {"internal": {}}}}
+
+    # Expose the direct output as internal-only data, this can be used by
+    # other externals.
+    data["tree"]["const"]["internal"]["packages"] = packages
+
+    # We also store all resolved packages and some meta information about
```

Comment:
This adds a very broad API to this otk external, I personally would start more targeted, we need the kernel name and version and in `images` the kernel is handled specially only. So maybe we could do the same here? Something like just expose a `const.kernel.{name,version}`? We will need to adjust `fragments/grub2.yam` to match the kernel and the string there includes `0-0.noarch` right now which also indicates that we need to tweak our tests to make this better testable.

Here is an alternative idea https://github.com/osbuild/otk/commit/5f378bb2631bc4cdb7cd6ae40d4cf1f0ec38b877

Reply:
We can set mock data to always be `0`, `0`, and `noarch` instead of the empty strings they are now.

I chose the broad API to not limit ourselves here also taking some inspiration from `osbuild-mpp` where it is currently possible to get the version of any package.

Reply:
Thanks, that is interesting! How is this used by folks that use `osbuild-mpp` - are there more use-cases then the kernel and if so, which ones? Would love to learn more here.

Reply:
The only call to the equivalent function in images that's not for looking up the kernel is to check if `dracut-config-rescue` exists in the package set, but it doesn't care about version.
https://github.com/osbuild/images/blob/67a2b2f65a2e0969039a3de17bf2c266390d01ab/pkg/manifest/os.go#L651